### PR TITLE
fix(lsp): gate node reads behind document writers

### LINKED ISSUE #698: node protocol: ingress-gate reads behind prior document writers
## What

The `kakehashi/node/*` method family is intentionally unclassified by `IngressOrderGate`, but its ID-based staleness checks only observe document/tracker mutations after the preceding writer handler has executed. In wire order `didClose` (or `didChange`) followed immediately by `kakehashi/node/text`, the node request can run first, see the still-present matching old incarnation, and return stale text rather than `null`. The same pre-writer window applies to navigation mints.

## Fix direction

Classify the full `kakehashi/node/*` family as per-document readers keyed by `params.textDocument.uri`, so they wait behind prior didOpen/didChange/didClose writer tickets. Add a deterministic gate test for didClose -> node/text and didChange -> node navigation. Reassess the existing comment claiming node self-staleness makes gating observably unnecessary.

## Scope

Found during fresh review of #569. Incarnation-tagged tracker entries close races after lifecycle mutations reach the store/tracker; they cannot observe a preceding wire-order writer that has not run yet. This is a pre-existing ingress-order contract and should be fixed separately.

### PR DESCRIPTION
**Pull request by atusy**
## Summary
- classify the full `kakehashi/node` method family as per-document ingress readers
- make node text/navigation wait behind earlier wire-order lifecycle writers
- add deterministic didClose and didChange ordering regressions

## Tests
- `cargo test --lib gate_runs_node`
- `cargo test --lib classify_routes_methods`
- `make test`
- `make check`
- `codex review --base origin/main`

Closes #698

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved request ordering for node-related operations.
  - Requests in the `kakehashi/node` and `kakehashi/node/*` family are now treated as gated readers, preventing them from proceeding ahead of prior document `didChange`/`didClose` events.
  - Updated and expanded tests to validate node text and children retrieval correctly waits behind preceding document updates.

### CODE REVIEW

File: src/lsp/ingress_order.rs [hunk 1]
```diff
@@ -818,6 +826,68 @@ mod tests {
...
+        assert!(node.poll().is_pending(), "node navigation remains gated");
+        release_tx.send(()).expect("didChange is waiting");
+        assert!(change.poll().is_ready());
+        assert!(node.is_woken());
+        assert!(node.poll().is_ready());
+        assert_eq!(
+            *log.lock().recover_poison("ingress_order::tests"),
+            vec!["change", "reader"]
+        );
+    }
```
[the remark this answers is not in the document]
Comment (atusy) [reply src/lsp/ingress_order.rs:889]: This is intentionally an ingress-gate unit test. It deterministically proves the navigation handler cannot observe pre-change state: polling it first stays pending until the preceding  handler completes, after which the log is . The gate does not interpret document contents or navigation responses, so adding a mock document state here would test a synthetic handler rather than provide additional evidence about the gate. The real handler’s state semantics are covered in its own layer; this regression covers the missing wire-order guarantee requested by #698.

File: src/lsp/ingress_order.rs [hunk 2]
```diff
@@ -818,6 +826,68 @@ mod tests {
...
+        assert!(node.poll().is_pending(), "node/text remains gated");
+        release_tx.send(()).expect("didClose is waiting");
+        assert!(close.poll().is_ready());
+        assert!(node.is_woken());
+        assert!(node.poll().is_ready());
+        assert_eq!(
+            *log.lock().recover_poison("ingress_order::tests"),
+            vec!["close", "reader"]
+        );
+    }
```
[the remark this answers is not in the document]
Comment (atusy) [reply src/lsp/ingress_order.rs:855]: The issue asks for a deterministic ingress-gate test, and this test directly proves that contract: the node/text inner handler is not invoked until the preceding didClose handler completes. The reader poll remains pending, then the completion log is close followed by reader. Observable null behavior belongs to the node handler and store integration; modeling another store inside this gate unit test would duplicate that subsystem without strengthening the ordering proof.

File: src/lsp/ingress_order.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (atusy) [reply src/lsp/ingress_order.rs:889]: For clarity: polling the node navigation future first remains pending until the preceding didChange handler completes, after which the log is change followed by reader. The gate does not interpret document contents or navigation responses, so a mock document state would test a synthetic handler rather than add evidence about the gate. This regression covers the missing wire-order guarantee requested by issue 698.
